microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/ls/AbstractMicroProfileDelegateCommandHandler.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.core.ls;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.jdt.ls.core.internal.IDelegateCommandHandler;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+
+import com.redhat.microprofile.jdt.core.IMicroProfilePropertiesChangedListener;
+import com.redhat.microprofile.jdt.internal.core.MicroProfilePropertiesListenerManager;
+
+/**
+ * Abstract class for MicroProfile JDT LS command handler
+ * 
+ * @author azerr
+ *
+ */
+public abstract class AbstractMicroProfileDelegateCommandHandler implements IDelegateCommandHandler {
+
+	private static final Logger LOGGER = Logger.getLogger(AbstractMicroProfileDelegateCommandHandler.class.getName());
+
+	/**
+	 * MicroProfile client commands
+	 */
+	private static final String MICROPROFILE_PROPERTIES_CHANGED_COMMAND = "microprofile/propertiesChanged";
+
+	private static final IMicroProfilePropertiesChangedListener LISTENER = (event) -> {
+		try {
+			// Execute client command with a timeout of 5 seconds to avoid blocking jobs.
+			JavaLanguageServerPlugin.getInstance().getClientConnection().executeClientCommand(

```

Comment:
should be getClientConnection().sendNotification(), since we're not waiting for a response

Reply:
Yes it should be that, but there is a bug on vscode-java. I tried to fix it with the following PR https://github.com/redhat-developer/vscode-java/pull/1368